hw/ip/entropy_src/data/entropy_src.hjson

```diff
@@ -143,10 +176,11 @@
              "excl:CsrAllTests:CsrExclAll"]
       fields: [
         { bits: "3:0",
-          name: "ENABLE",
+          name: "FIPS_ENABLE",
```

Comment:
How is `FIPS_ENABLE` different from `BOOT_BYPASS_DISABLE `? 

If both stages are enabled, does the FSM go from BOOT_INIT to FIPS mode?

Reply:
After this is done they will likely be completely orthongonal (and yes redundant).  

Currently there BOOT_BYPASS_ENABLE offers you two options:
1. Run though BOOT->STARTUP->CONTINUOUS modes
2. Skip boot to go through the sequence of STARTUP->CONTINUOUS

After this commit , Mark and I are gradually closing on what the implementation will look like, but here's the most likely outcome (as I understand it)
The FIPS_ENABLE switch will differentiate between two options:
1. If FIPS_ENABLE is enabled, the FSM goes trough the sequence: STARTUP->CONTINUOUS. This assumes that the full configuration has been applied.  (Note: there some debate as to whether BOOT is included in this sequence, as this is not to be used by the BOOTROM). 
 2. If FIPS_ENABLE is disabled, the FSM goes through the BOOT sequence, and _halts_.   It is then the ROM_EXT's job to properly configure the FIPS thresholds, and re-enable with the FIPS_ENABLE setting.   Previously there was nothing in the spec to prevent the creation of FIPS qualified seeds, but without any real FIPS thresholds (which we assume are not known by the BOOTROM).

Does that make sense?

Reply:
More discussion of this point in #9853

Reply:
Also note: this does not fully address #9853.  It just handles the messy business of changing the register IF and SW.

Reply:
@martin-lueker this is actually a very nice concise summary of the issue discussion. I am starting to lean toward option 1 to not have the boot phase. This removes a mode that will probably not be used, and one less combo to test. I would really like @moidx to endorse this though, as it does replace the current boot scheme, requiring SW intervention when it did not before.

Reply:
@mwbranstad when you say not having the boot phase, does that mean all entropy generation will be slower? even for the ROM case? let me know if I misunderstood. 

Based on what you guys are saying here, it does sound like having only `FIPS_ENABLE` where we use that to determine if we are in BOOT / STARTUP seems reasonable.  

I just want to make sure, when you say "halt", do you mean BOOT is only able to generate a certain amount of entropy? Because recall we've had issues in the past where even the startup cases needed a few rounds of EDN (this might be more about CSRNG / EDN) and wanted to make sure the halt doesn't prevent that.

Reply:
Thanks for the clarification. I think having FIPS separate from BOOT mode will yield a more ergonomic interface from the software perspective, so I agree with @mwbranstad and @martin-lueker on option1, and @tjaychen regarding collapsing the setting.

In the ROM we mainly need entropy to initialize countermeasures in the system, so we will probably end up updating `EDN0_BOOT_GEN_CMD` to ensure we have enough entropy bits available. We are also planning to add OTP settings for the BOOT (i.e. BYPASS) mode health thresholds.  Note also that software also uses entropy via `rv_core_ibex`. This type of entropy consumption is also needed in the ROM_EXT stage.

The earliest we can enable FIPS mode is ROM_EXT, and in some cases the enablement will be delayed until we reach later boot stages. The enablement needs to be done after we run KATs tests of the conditioner and DRBG.

Reply:
Thanks for the support, but clarify the two modes mentioned above:

> 1. If FIPS_ENABLE is enabled, the FSM goes trough the sequence: STARTUP->CONTINUOUS. This assumes that the full configuration has been applied.  (Note: there some debate as to whether BOOT is included in this sequence, as this is not to be used by the BOOTROM).
> 2. If FIPS_ENABLE is disabled, the FSM goes through the BOOT sequence, and _halts_.   It is then the ROM_EXT's job to properly configure the FIPS thresholds, and re-enable with the FIPS_ENABLE setting.   Previously there was nothing in the spec to prevent the creation of FIPS qualified seeds, but without any real FIPS thresholds (which we assume are not known by the BOOTROM).

Weren't meant as an either/or for implementation, rather two distinct modes of operation, selected by the new FIPS_ENABLE setting.    My apologies, but the references to an "Option 1" have me confused.

Reply:
yep that sounds good!

Reply:
> @mwbranstad when you say not having the boot phase, does that mean all entropy generation will be slower? even for the ROM case? let me know if I misunderstood.

I think what Mark is trying to say is that there would be no BOOT phase in the FIPS_ENABLE=True case, and _ONLY_ a BOOT in the FIPS_ENABLE=False case

> Based on what you guys are saying here, it does sound like having only `FIPS_ENABLE` where we use that to determine if we are in BOOT / STARTUP seems reasonable.

I prefer this also.  At this point the fewer switches the better.

> I just want to make sure, when you say "halt", do you mean BOOT is only able to generate a certain amount of entropy? Because recall we've had issues in the past where even the startup cases needed a few rounds of EDN (this might be more about CSRNG / EDN) and wanted to make sure the halt doesn't prevent that.

The CSRNG could generate all the entropy it wants, but it would not be able to perform a reseed.  So I guess the question is, how long do we want to go with out a reseed?

Reply:
@tjaychen I think @martin-lueker answered all of the questions that you asked, but more on the last one:
Boot-Req mode is being used to fetch non-FIPS entropy for EDN0 for BOOT. The number of genbits returned needs to be programmed however(but there is a default of ~1K). Is the boot plan to match the planned endpoint consumption of entropy exactly to the csrng GEN command, or just reset csrng (disable) once the BOOTROM phase is over (which should also work)? Point being that GEN commands must complete cleanly before a new command can be sent.

Reply:
that sounds good. Yeah I dont at the moment see a need for a reseed.. so it feels like if we just have CSRNG pump out more than should be fine.

Reply:
Hi @martin-lueker sorry about the confusion. What you said sounds good!